fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java

```diff
@@ -867,7 +871,53 @@ public Scope visitTableFunction(TableFunctionRelation node, Scope scope) {
             if (names != null && !names.isEmpty()) {
                 namesArray = names.toArray(String[]::new);
             }
-            Function fn = Expr.getBuiltinFunction(node.getFunctionName().getFunction(), argTypes, namesArray,
+
+            String functionName = node.getFunctionName().getFunction();
+            if (FunctionSet.GENERATE_SERIES.equalsIgnoreCase(functionName) && args.size() >= 2) {
+                int stepParamIndex = 2;
+                List<Integer> startStopParamIndex = new ArrayList<>();
+                if (namesArray != null) {
+                    for (int i = 0; i < namesArray.length; i++) {
+                        if ("step".equalsIgnoreCase(namesArray[i])) {
+                            stepParamIndex = i;
+                        } else {
+                            startStopParamIndex.add(i);
+                        }
+                    }
+                }
+
+                boolean isGenerateTimeSeries = true;
+                for (Integer index : startStopParamIndex) {
+                    if (!(argTypes[index].isDatetime() || Type.canCastTo(argTypes[index], Type.DATETIME)) ||
+                            args.get(index) instanceof NullLiteral) {
+                        isGenerateTimeSeries = false;
+                    }
+                }
+                isGenerateTimeSeries = !Arrays.stream(argTypes).allMatch(Type::isFixedPointType) && isGenerateTimeSeries;
+
```

Comment:
Could you explain the logic of this line?

Reply:
because FixedPointType can be cast to DataTime, this can prevent expr without step(eg generate_series(1, 3)) convert to generate_series_datatime